production/ksonnet/loki/config.libsonnet

```diff
@@ -132,10 +157,16 @@
             },
           },
         },
+      } + if $._config.queryFrontend.sharded_queries_enabled then {
+        parallelise_shardable_queries: true,
+      } else {},
+      querier: {
+        query_ingesters_within: '2h', // twice the max-chunk age (1h default) for safety buffer
       },
       limits_config: {
         enforce_metric_name: false,
-        max_query_parallelism: 32,
+        // align middleware parallelism with shard factor to optimize one-legged sharded queries.
+        max_query_parallelism: $._config.queryFrontend.shard_factor,
```

Comment:
If you do this, it means splitting by time is not useful anymore as they will execute sequentially after sharding.

Reply:
Assuming we always run all shard in //.